### PR TITLE
Allow some rng functions to be called on the GPU

### PR DESCRIPTION
**Pull request by alexrlongne**
### Background

* KT recently reworked the draco build system to better support CUDA builds. Part of that work exposed the `GPU_HOST_DEVICE` macro in `device/config.h` that allows functions to be decorated with `__host__ __device__` only when CUDA builds are on. Because this was not previously available in draco, I had a copied some `rng` files to Jayenne to decorate them for GPU execution. This PR uses the macros in `device/config.h` to mark functions in `rng` needed on the GPU. After this PR is merged I'll delete the copied `rng` files in Jayenne.

### Purpose of Pull Request

* [Part of Redmine Issue #1814](https://rtt.lanl.gov/redmine/issues/1814)

### Description of changes

* Decorate RNG functions with GPU host device calls
* Add device dependency to RNG
* Move RNG to Level 3 packages

### Status

* Reference:
  * [Pre-Merge Code Review](https://github.com/lanl/Draco/wiki/Style-Guide)
  * [CDash Status](https://rtt.lanl.gov/cdash/index.php?project=Draco&filtercount=1&showfilters=1&field1=buildname&compare1=63&value1=-pr)
* Checks
  * [x] Travis/Appveyor CI checks pass
  * [x] Code coverage does not decrease
  * [x] [Valgrind test passes](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] [Toss3 checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] [Trinitite checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] Code reviewed/approved, sufficient DbC checks, testing, documentation

### CODE REVIEW

File: src/rng/Counter_RNG.hh [hunk 1]
```diff
@@ -11,6 +11,7 @@
 #ifndef Counter_RNG_hh
 #define Counter_RNG_hh
 
+#include "device/config.h"
 #include "rng/config.h"
```
Comment (KineticTheory) [inline src/rng/Counter_RNG.hh]: Does `rng/config.h` depend on the content of `device/config.h`? If so, I recommend moving the include from this location and into `rng/config.h`.

File: src/rng/Rnd_Control_Inline.hh [hunk 2]
```diff
@@ -11,6 +11,7 @@
 #define rtt_rng_Rnd_Control_Inline_hh
 
 #include "Counter_RNG.hh"
+#include "device/config.h"
```
Comment (KineticTheory) [inline src/rng/Rnd_Control_Inline.hh]: FYI - This include is not required.  `Counter_RNG.hh` has already included `device/config.h`.

File: src/rng/Counter_RNG.hh (hunk 1 above)
Comment (alexrlongne) [reply src/rng/Counter_RNG.hh]: Nope. I don't use anything from `device/config.h` in `rng/config.h`

File: src/rng/Rnd_Control_Inline.hh (hunk 2 above)
Comment (alexrlongne) [reply src/rng/Rnd_Control_Inline.hh]: Thanks! Removing...

**Comment by alexrlongne** [comment]
@KineticTheory I removed the extra include line. CI is rerunning.

**Pull request merged**